lib/spack/spack/test/cmd/view.py

```diff
@@ -0,0 +1,91 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack.main import SpackCommand
+
+activate = SpackCommand('activate')
+extensions = SpackCommand('extensions')
+install = SpackCommand('install')
+view = SpackCommand('view')
+
+
+def test_view_external(
+        tmpdir, builtin_mock, mock_archive, mock_fetch, config,
+        install_mockery):
+    install('externaltool')
+    viewpath = str(tmpdir.mkdir('view'))
+    output = view('symlink', viewpath, 'externaltool')
+    assert 'Skipping external package: externaltool' in output
+
+
+def test_view_extension(
+        tmpdir, builtin_mock, mock_archive, mock_fetch, config,
+        install_mockery):
+    install('extendee')
+    install('extension1@1.0')
+    install('extension1@2.0')
+    install('extension2@1.0')
+    viewpath = str(tmpdir.mkdir('view'))
+    view('symlink', viewpath, 'extension1@1.0')
```

Comment:
IMO it is also worth confirming that the files in the `extension1` package were created in the view prefix.